src/main/java/org/elasticsearch/index/query/TermQueryParser.java

```diff
@@ -80,27 +74,19 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
             }
             parser.nextToken();
         } else {
-            value = parser.text();
+            value = parser.objectText();
```

Comment:
we discussed this and we went for always keeping `BytesRef` within `TermQueryBuilder` rather than mixing up `String` and `BytesRef` depending on whether we received the query via java api or we parsed it through json. Java api will still have a setter that accepts a string, but internally that setter will convert from `String` to `BytesRef`.

We do have to fix the above `parser.text()` which looks like a bug that never manifested as when parsing and executing  on the same (data) node everything still works. But we are currenly parsing the value in the term query short format (`term: { field: value}`) always as a string.